api/src/utilities/s3-helpers.ts

```diff
@@ -0,0 +1,92 @@
+import { InternalServerErrorException } from '@nestjs/common';
+import { GetObjectCommand, S3Client } from '@aws-sdk/client-s3';
+import { createReadStream } from 'fs';
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
+import { Upload } from '@aws-sdk/lib-storage';
+
+export const checkArgs = (args: Record<string, string>): void => {
+  Object.keys(args).forEach((key) => {
+    if (!args[key]) {
+      throw new InternalServerErrorException(
+        `S3 env variable ${key} is missing`,
+      );
+    }
+  });
+};
+
+/**
+ * @param bucket the s3 bucket the object lives inside of
+ * @param key the key for the object we want a presigned url for
+ * @param region the region the s3 bucket exists in. Specified in AWS
+ * @returns the presigned url for downloading a file
+ */
+export const generatePresignedGetURL = async (
+  bucket: string,
+  key: string,
+  region: string,
+): Promise<string> => {
+  checkArgs({
+    bucket,
+    region,
+  });
+
+  const client = new S3Client({
+    region,
+  });
+
+  const url = await getSignedUrl(
+    client,
+    new GetObjectCommand({
+      Bucket: bucket,
+      Key: key,
+    }),
+    {
+      expiresIn: 1000 * 60 * 5, // expires 5 minutes after generation
```

Comment:
nitpick: Lets make this configurable but have a default of 5 minutes if it's not set. I wouldn't be surprised if Doorway wants us to fine tune this and it would be nice to not have to make a code change

Reply:
added!